libs/angular-integration-interface/src/lib/services/workspace.service.spec.ts

```diff
@@ -0,0 +1,454 @@
+import { TestBed } from '@angular/core/testing'
+import { WorkspaceService } from './workspace.service'
+import { of } from 'rxjs'
+import { AppStateService } from './app-state.service'
+
+describe('WorkspaceService', () => {
+  let service: WorkspaceService
+  let mockAppStateService: any
+  const params: Record<string, unknown> = {
+    "id": 5,
+    "key":"xy"
+  };
+
+  const paramsWrong: Record<string, unknown> = {
+    "idx": 5
+  };
+
+  beforeEach(() => {
+    mockAppStateService = {
+      currentWorkspace$: of({
+        baseUrl: 'http://example.com',
+        routes: [
+          {
+            appId: 'onecx-workspace-ui',
+            productName: 'onecx-workspace',
+            baseUrl: 'http://example.com/workspace/baseurl',
+            endpoints: [
+              { name: 'details', path: '/details/{id}' },
+              { name: 'edit', path: '[[details]]' },
+              { name: 'change', path: '[[edit]]' },
+            ],
+          },
+        ],
+      }),
+    }
+
+    TestBed.configureTestingModule({
+      providers: [WorkspaceService, { provide: AppStateService, useValue: mockAppStateService }],
+    })
+
+    service = TestBed.inject(WorkspaceService)
+  })
+
+  it('should be created', () => {
+    expect(service).toBeTruthy()
+  })
+
+  //Testcases
+  //1 Best case scenario
+  //2 workspace baseUrl not there
+  //3 workspace baseUrl is empty
+  //4 workspace has no routes
+  //5 workspace has emtpy routes
+  //6 route was not found by appId and productName
+  //7 workspace.route baseUrl not there
+  //8 workspace.route baseUrl is empty
+  //9 endpoints are empty
+  //10 endpoint with name was not found
+  //11 test endpoint with 1 alias
+  //12 test endpoint with 2 alias
+  //13 test endpoint with wrong alias
+  //14 param was not found
+  //15 no params given as param of getUrl
+  //16 2 params
+  //17 no endpoints available
+  //18 test getUrl with double //
+
+  //1
+  it('should find endpoint and return an observable that emits correct url ', (done) => {
```

Comment:
It is a good idea to wrap all tests which are testing the same method into a sub-describe where the description is the name of the function. So it is easy to add tests for new/additional methods later